vic/extensions/rout_rvic/include/rout.h

```diff
@@ -0,0 +1,88 @@
+
+#include <vic_def.h>
+
+/******************************************************************************
+ * @brief   Routing Structs
+ *****************************************************************************/
+typedef struct {
+    size_t full_time_length;                  /*scalar - total number of timesteps*/
+    size_t nTimesteps;                        /*scalar - number of timesteps*/
+    size_t nSources;                          /*scalar - number of sources*/
+    size_t nOutlets;                          /*scalar - length of subset*/
+    size_t *source2outlet_ind;                /*1d array - source to outlet mapping*/
+    int *source_y_ind;                        /*1d array - source y location*/
+    int *source_x_ind;                        /*1d array - source x location*/
+    double *source_lat;                       /*1d array - Latitude coordinate of source grid cell*/
+    double *source_lon;                       /*1d array - Longitude coordinate of source grid cell*/
+    double *outlet_lat;                       /*1d array - Latitude coordinate of outlet grid cell*/
+    double *outlet_lon;                       /*1d array - Longitude coordinate of outlet grid cell*/
+    int *source_VIC_index;                    /*1d array - mapping of routing-source index to VIC index*/
+    int *outlet_VIC_index;                    /*1d array - mapping of routing-outlet index to VIC index*/
+    int *source_time_offset;                  /*1d array - source time offset*/
+    double *unit_hydrograph;                  /*2d array[times][sources] - unit hydrographs*/
+    double *aggrunin;                         /*2d array[ysize][xsize] - vic runoff flux*/
+} rout_param_struct;
+
+/******************************************************************************
+ * @brief   main routing Struct
+ *****************************************************************************/
+typedef struct {
+    char param_filename[MAXSTRING];
+    rout_param_struct rout_param;
+    double *ring;
+    double *discharge;
+} rout_struct;
+
+/******************************************************************************
+ * @brief   Function prototypes for the rout_rvic extension
+ *****************************************************************************/
+void rout_start(void);      // read global parameters for routing
+void rout_alloc(void);      // allocate memory
+void rout_init(void);       // initialize model parameters from parameter files
+void rout_run(void);        // run routing over the domain
+void rout_write(void);      // write routine for routing
+void rout_finalize(void);   // clean up routine for routing
+
+/******************************************************************************
+ * @brief   MPI Function prototypes for the rout_rvic extension
+ *****************************************************************************/
+void get_scatter_var_double(double *, double *);
+void gather_put_var_double(double *, double *);
```

Comment:
I don't think so, because we use scatter and gather for variables and for for netcdf files. However I can move these functions to the main VIC-code if required.